homeassistant/components/roborock/button.py

```diff
@@ -115,13 +117,26 @@ def __init__(
         super().__init__(
             f"{entity_description.key}_{coordinator.duid_slug}",
             coordinator.device_info,
-            coordinator.api,
+            api=coordinator.properties_api.command,
         )
         self.entity_description = entity_description
+        self._consumable = coordinator.properties_api.consumables
 
     async def async_press(self) -> None:
         """Press the button."""
-        await self.send(self.entity_description.command, self.entity_description.param)
+        try:
+            await self._consumable.reset_consumable(self.entity_description.attribute)
+        except RoborockException as err:
+            # This error message could be improved since it is fairly low level
+            # and technical. Can add a more user friendly message with the
+            # name of the attribute being reset.
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="command_failed",
+                translation_placeholders={
+                    "command": "RESET_CONSUMABLE",
+                },
+            ) from err
```

Comment:
I would almost suggest to not use a translated one for now and translate when we figured this one out, this way we can tinker without putting unneeded pressure on the translators

Reply:
Ah check, no I saw the discussion on creating a better error to raise, and I think there is room for improvement here but I also didn't want this discussion to block the rest